teuthology/dispatcher/supervisor.py

```diff
@@ -0,0 +1,270 @@
+import logging
+import os
+import subprocess
+import time
+import yaml
+
+from datetime import datetime
+
+import teuthology
+from teuthology import report
+from teuthology import safepath
+from teuthology.config import config as teuth_config
+from teuthology.exceptions import SkipJob
+from teuthology import setup_log_file, install_except_hook
+from teuthology.lock.ops import reimage_many
+from teuthology.misc import get_user, archive_logs, compress_logs
+from teuthology.config import FakeNamespace
+from teuthology.job_status import get_status
+from teuthology.nuke import nuke
+from teuthology.kill import kill_job
+from teuthology.task.internal import add_remotes
+from teuthology.misc import decanonicalize_hostname as shortname
+from teuthology.lock import query
+
+log = logging.getLogger(__name__)
+
+
+def main(args):
+
+    verbose = args["--verbose"]
+    archive_dir = args["--archive-dir"]
+    teuth_bin_path = args["--bin-path"]
+    config_file_path = args["--job-config"]
+
+    with open(config_file_path, 'r') as config_file:
+        job_config = yaml.safe_load(config_file)
+
+    loglevel = logging.INFO
+    if verbose:
+        loglevel = logging.DEBUG
+    log.setLevel(loglevel)
+
+    log_file_path = os.path.join(job_config['archive_path'],
+                                 f"supervisor.{job_config['job_id']}")
+    setup_log_file(log_file_path)
+    install_except_hook()
+
+    # reimage target machines before running the job
+    if 'targets' in job_config:
+        reimage_machines(job_config)
+        with open(config_file_path, 'w') as f:
+            yaml.safe_dump(job_config, f, default_flow_style=False)
+
+    try:
+        run_job(
+            job_config,
+            teuth_bin_path,
+            archive_dir,
+            verbose
+        )
+    except SkipJob:
+        return
+
+
+def run_job(job_config, teuth_bin_path, archive_dir, verbose):
+    safe_archive = safepath.munge(job_config['name'])
+    if job_config.get('first_in_suite') or job_config.get('last_in_suite'):
+        suite_archive_dir = os.path.join(archive_dir, safe_archive)
```

Comment:
I suppose this is job archive directory not a suite one.

Reply:
Ah sorry, changed it to `job_archive`.